src/switchcraft_advanced/services/intune_service.py

```
@@ -0,0 +1,362 @@
+            missing = [r for r in mandatory if r not in roles]
+            if missing:
+                return False, f"Missing Mandatory Role: {', '.join(missing)}"
+
+            # Check optional (accept higher/alternative if logic allows, here we check Group Read)
+            # Accept Group.Read.All OR Group.ReadWrite.All
+            has_group_read = any(r in roles for r in ["Group.Read.All", "Group.ReadWrite.All"])
+
+            if not has_group_read:
+                return True, "ok_with_warning: Missing 'Group.Read.All'. App assignment verification will fail, but upload will work."
+
+            return True, "OK"
```

Ruff (0.14.8)

Consider moving this statement to an `else` block

(TRY300)